### DISCUSSION TITLE
pigallery in docker behind nginx reverse proxy within a ddns accessible NAS box

### DISCUSSION THREAD

**Discussion by NoBurpees** [bpatrik/pigallery2#767] [Q&A]
Hi,

first thanx for this great app, I've been looking for a directory base image browser and I'm very glad that it exists here ;-)

Unfortunately I've been banging my head against various table tops for some nights, since I can't get pigallery2 working from within docker, behind a  nginx rev proxy and via a ddns connected NAS box.

I'm running it as a docker container with a port mapping 9881:80 and can access it easily from within my LAN (bypassing nginx) via http://here.at.home:9881  (here.at.home being my NAS)

So anything seems to be working and I got to step 2 - reverse proxy:

location /pigallery2 {
      proxy_pass              http://localhost:9881;
      proxy_http_version 1.1;
      proxy_set_header Upgrade $http_upgrade;
      proxy_set_header Connection 'upgrade';
      proxy_set_header Host $host;
      proxy_cache_bypass $http_upgrade;
}

Trying to access via internet/DDNS doesn't work - get a time-out.

So being not the top pro regarding nginx reverse proxy config I'm stuck here.

Can someone please share their nginx config which lets them connect to the docker run pigallery2 via a rev proxy? What am I doing wrong here?

Best regards
Mark.

**Comment by rorymurtagh** [comment]
If PiGallery2 is running on http://here.at.home:9881 then the nginx proxy_pass would be to http://here.at.home:9881. This is unless 'here.at.home' is localhost but in your case I think that's not the case.
